verifiers/internal/gha/slsaprovenance/slsaprovenance.go

```diff
@@ -0,0 +1,73 @@
+package slsaprovenance
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	dsselib "github.com/secure-systems-lab/go-securesystemslib/dsse"
+	serrors "github.com/slsa-framework/slsa-verifier/v2/errors"
+)
+
+type Provenance interface {
+	// BuilderID returns the builder id in the predicate.
+	BuilderID() string
+
+	// SourceURI is the full URI (including tag) of the source material.
+	SourceURI() (string, error)
+
+	// ConfigURI is the full URI (including tag) of the configuration material.
+	ConfigURI() string
+
+	// Subject is the list of intoto subjects in the provenance.
+	Subjects() []intoto.Subject
+
+	// GetStringFromEnvironment retrieves a string parameter from the environment
+	// attested to in the provenance.
+	GetStringFromEnvironment(name string) (string, error)
```

Comment:
do you want to go further and have the high-level functions `Branch()`, `Tag()`?

Reply:
Yeah, I was a little bit unsure. On one hand it makes it very clear what info we require from the provenance. On the other hand I wasn't too sure to optimize away all of this too early. 

I will follow-up w/ an issue and mark as fixed when I add the v1.0 impl

Reply:
https://github.com/slsa-framework/slsa-verifier/issues/464

Reply:
SG!

Reply:
FWIW i'm leaning towards exposing the high-level functions. Addressing in my next follow-up!
Thanks!

Reply:
sure, no urgency on this